### PR TITLE
shared: Add preview

### LINKED ISSUE #1264: Add preview to shared ip
Requested at [WT:TW](en.wikipedia.org/wiki/Special:Permalink/998064916#Preview_function):

>The preview function, that is quite handy and found in all of the welcome and warning templates is oddly missing from the SharedIP templates (and all of it's the subs, such as DynamicIP, StaticIP, Edu, Gov't, etc.) Was that deliberate? Is it possible to add that function to those templates?

### PR DESCRIPTION
**Pull request by TheTVExpert**
Closes #1264

### CODE REVIEW

**Review by Amorymeltzer — CHANGES_REQUESTED** [review]
@TheTVExpert I actually think this should use the pop-up style preview, rather than the in-line preview, since these are all huge templates.  It shouldn't be too complicated, there's a good example in the welcome module.

**Comment by TheTVExpert** [comment]
@Amorymeltzer Done.

**Review by Amorymeltzer — CHANGES_REQUESTED** [review]
Just a couple minor suggestions about window size/style, but otherwise LGTM!  Appreciate the speedy call-to-action! 👍

**Comment by TheTVExpert** [comment]
@Amorymeltzer Done.

File: modules/friendlyshared.js [hunk 1]
```diff
@@ -165,6 +165,47 @@ Twinkle.shared.callbacks = {
 	}
 };
 
+Twinkle.shared.preview = function(form) {
+	var input = Morebits.quickForm.getInputData(form);
+	if (input.template !== undefined) {
```
[the remark this answers is not in the document]
Comment (Amorymeltzer) [reply modules/friendlyshared.js]: Whoops, `input.template`, my bad!

**Comment by Amorymeltzer** [comment]
LGTM!  Merged as 7a8e2a9

**Pull request closed without merging**